### PR TITLE
added types for katex sub module katex2tex

### PR DESCRIPTION
**Pull request by s-weigand**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/KaTeX/KaTeX/blob/master/contrib/copy-tex/katex2tex.js
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

**Also**:
I wanted to mention that i had some problems with the prettier formating in the git hook. It kept reformating and added additional new lines, on windows. 
If I manually (vs-code prettier plugin) format the file (on ubuntu this time), `precise-commits` even breaks the code since it runs on changed lines only.
And it used 2 space indentation instead of 4 space indentation in `types/katex/katex-tests.ts`. Which can be fixed by adding `"tabWidth": 4,` back to `.prettierrc.json`, which was removed by #37400 .

### CODE REVIEW

File: types/katex/contrib/katex2tex.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+declare module 'katex/contrib/copy-tex/katex2tex' {
+    /**
+     * Set these to how you want inline and display math to be delimited.
+     */
+    interface ICopyDelimiters {
```
Comment (dreamerblue) [inline types/katex/contrib/katex2tex.d.ts]: Recommend export param interface and named without 'I'.
Such as `export interface CopyDelimiters {`.

**Review by dreamerblue — CHANGES_REQUESTED** [review]
See review comment in
- `types/katex/contrib/katex2tex.d.ts`

**Comment by demensky** [comment]
> And it used 2 space indentation instead of 4 space indentation in types/katex/katex-tests.ts. Which can be fixed by adding "tabWidth": 4, back to .prettierrc.json, which was removed by #37400 .

[Prettier supports EditorConfig](https://prettier.io/blog/2017/12/05/1.9.0.html#add-editorconfig-support-3255-https-githubcom-prettier-prettier-pull-3255-by-josephfrazier-https-githubcom-josephfrazier).

If you run prettier manually, then everything will be fine.

```sh
prettier --write types/katex/katex-tests.ts
```

I think the problem is `precise-commits` ([last commit was almost 1.5 years ago](https://github.com/nrwl/precise-commits/issues/49#issue-482017386)).

@orta what do you think of this problem?

**Comment by orta** [comment]
Perhaps there is a hierarchy of editor config's on one computer but not the other? ( e.g. as it looks backwards through the fs for any configs one takes precedence ) - the use of precise commits is to migrate slowly to a place where it's feasible to run prettier over the whole codebase eventually. 

If it's getting in the way, skip it with `--no-verify`, nothing will break on the DTS side 👍

**Comment by s-weigand** [comment]
@demensky I had also suspected `precise-commits` as being the culprit, since many people have formatting problems (issues on their repo).

@orta Thanks for the tipp, `--no-verify solved my formatting problem. Maybe this can be added to the contributing guide.

**Review by dreamerblue — APPROVED** [review]
ok

**Pull request merged**